coordinator/src/main/scala/com/socrata/datacoordinator/service/Main.scala

```diff
@@ -449,13 +449,10 @@ object Main extends DynamicPortMap {
 
     PropertyConfigurator.configure(Propertizer("log4j", serviceConfig.logProperties))
 
-    val secondaries: Set[String] = serviceConfig.secondary.groups.flatMap(_._2.instances).toSet
+    val secondaries: Set[String] = serviceConfig.secondary.groups.flatMap(_._2.instances.keySet).toSet
+    // TODO: remove this
     val secondariesNotAcceptingNewDatasets: Set[String] =
-      serviceConfig.secondary.groups.flatMap(_._2.instancesNotAcceptingNewDatasets).flatten.toSet
-    serviceConfig.secondary.groups.foreach { case (name, group) =>
-      if (!group.instancesNotAcceptingNewDatasets.getOrElse(Set.empty).subsetOf(group.instances))
-        throw new Exception(s"For secondary group $name instancesNotAcceptingNewDatasets must be a subset of instances!")
-    }
+      serviceConfig.secondary.groups.flatMap(_._2.instances.filter(!_._2.acceptingNewDatasets).keySet).toSet
```

Comment:
The anonymous underscore -> tuple makes this pretty tough to parse. Probably worth unwrapping the tuples.